pandas/core/index.py

```diff
@@ -973,7 +976,9 @@ def _convert_list_indexer(self, keyarr, kind=None):
         and we have a mixed index (e.g. number/labels). figure out
         the indexer. return None if we can't help
         """
-        if (kind is None or kind in ['iloc','ix']) and (is_integer_dtype(keyarr) and not self.is_floating()):
+        if (kind is None or kind in ['iloc', 'ix']) and (
+            is_integer_dtype(keyarr) and not self.is_floating() and not isinstance(keyarr, ABCPeriodIndex)):
```

Comment:
though you might simply be able to define ``_convert_list_indexer`` in ``PeriodIndex`` to handle this in a nicer way